test/rateLimiters/tokenBucket.test.ts

```diff
@@ -0,0 +1,81 @@
+import TokenBucket from '../../src/rateLimiters/tokenBucket';
+
+const CAPACITY = 10;
+const REFILL_RATE = 1; // 1 token per second
+
+let limiter: TokenBucket;
+let user1;
+let user2;
+let user3;
+
+xdescribe('Test token bucket functionality', () => {
+    beforeAll(() => {
+        user1 = '1';
+        user2 = '2';
+        user3 = '3';
+    });
+
+    beforeEach(() => {
+        // Initialize a new token bucket before each test
+        // create a mock user
+        // intialze the token bucket algorithm
+        limiter = new TokenBucket(CAPACITY, REFILL_RATE);
+    });
+
+    test('allows a user to consume up to their current allotment of tokens', () => {
+        // Test 1 token requested
+        expect(limiter.processRequest(user1, 1)).toBe(true);
+        // Test < CAPACITY tokens requested
+        expect(limiter.processRequest(user2, CAPACITY - 1)).toBe(true);
+        // <= CAPACITY tokens requested
+        expect(limiter.processRequest(user3, CAPACITY)).toBe(true);
+
+        setTimeout(() => {
+            // make sure user doesn't get extra tokens
+            expect(limiter.processRequest(user1, CAPACITY + 1)).toBe(false);
+        }, 1000);
+    });
+
+    test("blocks requets exceeding the user's current allotment of tokens", () => {
+        // Test > capacity tokens reqeusted
+        expect(limiter.processRequest(user1, CAPACITY + 1)).toBe(false);
+        // allowed to take full amount
+        expect(limiter.processRequest(user1, CAPACITY)).toBe(true);
```

Comment:
The test for "blocks requets exceeding the user's current allotment of tokens" on line 39 is technically dependent on the algorithm working to allow users to consume their allotment of tokens. It is a dependency on something you aren't testing for. It depends on how picky we want to be with tests.